### PR TITLE
PMM-7656 go modules

### PR DESCRIPTION
**Pull request by Dasio**
https://jira.percona.com/browse/PMM-7656
FB: https://github.com/Percona-Lab/pmm-submodules/pull/1604

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -22,14 +22,15 @@ LD_FLAGS = -ldflags " \
...
+	go build -modfile=tools/go.mod -o bin/goimports golang.org/x/tools/cmd/goimports
 
-	go install ./vendor/github.com/quasilyte/go-consistent \
-				./vendor/golang.org/x/tools/cmd/goimports
-
-	go test -i ./...
-	go test -race -i ./...
+	go install ./...
+	go test ./...
+	go test -race ./...
```
Comment (artemgavrilov) [inline Makefile]: Do you know why we run tests here twice? It looks like `go test -race ./...` is enougth

File: .travis.yml [hunk 2]
```diff
@@ -45,9 +45,6 @@ before_install:
   - docker-compose up -d
 
 install:
-  # ensure that vendor/ is in sync with code and Gopkg.toml/lock
-  - curl https://raw.githubusercontent.com/golang/dep/master/install.sh | sh
-  - dep check
   # install reviewdog and golangci-lin
   - curl https://raw.githubusercontent.com/reviewdog/reviewdog/master/install.sh| sh -s
```
Comment (artemgavrilov) [inline .travis.yml]: It seems that you don't need it anymore. `make init` will install both tools.

File: Makefile [hunk 3]
```diff
@@ -22,14 +22,14 @@ LD_FLAGS = -ldflags " \
...
+	go build -modfile=tools/go.mod -o bin/reviewdog github.com/reviewdog/reviewdog/cmd/reviewdog
+	go build -modfile=tools/go.mod -o bin/goimports golang.org/x/tools/cmd/goimports
 
-	go install ./vendor/github.com/quasilyte/go-consistent \
-				./vendor/golang.org/x/tools/cmd/goimports
-
-	go test -i ./...
-	go test -race -i ./...
+	go install ./...
+	go test -race ./...
```
Comment (artemgavrilov) [inline Makefile]: Oh, I got it. 
```
        -i
            Install packages that are dependencies of the test.
            Do not run the test.
            The -i flag is deprecated. Compiled packages are cached automatically.
```

So it was just dependency resolving, but anyway, why twice?

IMHO we don't need `go test` here, and probably we don't need `go install`

Comment (artemgavrilov) [reply Makefile]: @Dasio ^

Comment (Dasio) [reply Makefile]: Sure, go install was needed because of go test was using built binary.
If i remove test from here, it's fine

**Review by artemgavrilov — APPROVED** [review]
LGTM 👍

Let's fix `tip` builds in this repo, by adding a new `build` tag (but keeping the old one) https://golang.org/doc/go1.16#go/build/constraint
```
//go:build tools
// +build tools
```

**Review by oter — APPROVED** [review]

**Pull request merged**